.ci/pytorch/test.sh

```diff
@@ -18,6 +18,10 @@ BUILD_DIR="build"
 BUILD_RENAMED_DIR="build_renamed"
 BUILD_BIN_DIR="$BUILD_DIR"/bin
 
+#Set Default values for these variables in case they are not set
+SHARD_NUMBER="${SHARD_NUMBER:=1}"
+NUM_TEST_SHARDS="${NUM_TEST_SHARDS:=1}"
```

Comment:
Why this is needed? I would prefer it to error out if SHARD_NUMBER is not defined

Reply:
We added these to simplify our internal testing.

Reply:
In order to run inductor config, it seems that SHARD_NUMBER is required now. (https://github.com/pytorch/pytorch/pull/110544/files#diff-9709e5db13aeac90c0312b2b8da34b37cf51242ec789ca959fcf1ba295a8da7aR1101)

Although this variable is always defined in all the CI  yaml files that trigger these tests, it's nice to be able to run this file locally outside of CI context.